### PR TITLE
refactor: remove default auto-explode for map_many_private

### PR DESCRIPTION
**Pull request by reswqa**
per comment(https://github.com/pola-rs/polars/pull/11230#discussion_r1333920358).

Note: I just keep the original value for `auto-explode`, doesn't change any setting. But this may be a good opportunity to check if our previous setting for `auto-explode` was correct. 😆

### CODE REVIEW

File: crates/polars-plan/src/dsl/dt.rs [hunk 1]
```diff
@@ -265,14 +266,15 @@ impl DateLikeNameSpace {
...
-            .map_many_private(FunctionExpr::DateOffset, &[by], false)
+            .map_many_private(FunctionExpr::DateOffset, &[by], true, false)
     }
 
     #[cfg(feature = "timezones")]
     pub fn replace_time_zone(self, time_zone: Option<TimeZone>, ambiguous: Expr) -> Expr {
         self.0.map_many_private(
             FunctionExpr::TemporalExpr(TemporalFunction::ReplaceTimeZone(time_zone)),
             &[ambiguous],
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/dt.rs:277]: `false`

File: crates/polars-plan/src/dsl/dt.rs [hunk 2]
```diff
@@ -265,14 +266,15 @@ impl DateLikeNameSpace {
     #[cfg(feature = "date_offset")]
     pub fn offset_by(self, by: Expr) -> Expr {
         self.0
-            .map_many_private(FunctionExpr::DateOffset, &[by], false)
+            .map_many_private(FunctionExpr::DateOffset, &[by], true, false)
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/dt.rs:269]: `false`

File: crates/polars-plan/src/dsl/dt.rs [hunk 3]
```diff
@@ -281,6 +283,7 @@ impl DateLikeNameSpace {
         self.0.map_many_private(
             FunctionExpr::TemporalExpr(TemporalFunction::Combine(tu)),
             &[time],
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/dt.rs:286]: `false`

File: crates/polars-plan/src/dsl/list.rs [hunk 4]
```diff
@@ -98,6 +102,7 @@ impl ListNameSpace {
         self.0.map_many_private(
             FunctionExpr::ListExpr(ListFunction::Take(null_on_oob)),
             &[index],
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/list.rs:105]: `false`

File: crates/polars-plan/src/dsl/list.rs [hunk 5]
```diff
@@ -119,6 +124,7 @@ impl ListNameSpace {
         self.0.map_many_private(
             FunctionExpr::ListExpr(ListFunction::Join),
             &[separator],
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/list.rs:127]: `false`

File: crates/polars-plan/src/dsl/list.rs [hunk 6]
```diff
@@ -169,6 +175,7 @@ impl ListNameSpace {
         self.0.map_many_private(
             FunctionExpr::ListExpr(ListFunction::Slice),
             &[offset, length],
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/list.rs:178]: `false`

File: crates/polars-plan/src/dsl/list.rs [hunk 7]
```diff
@@ -278,11 +287,11 @@ impl ListNameSpace {
             .map_many_private(
                 FunctionExpr::ListExpr(ListFunction::SetOperation(set_operation)),
                 &[other],
+                false,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/list.rs:290]: `false`

File: crates/polars-plan/src/dsl/string.rs [hunk 8]
```diff
@@ -119,13 +123,17 @@ impl StringNameSpace {
     /// Extract each successive non-overlapping match in an individual string as an array
     pub fn extract_all(self, pat: Expr) -> Expr {
         self.0
-            .map_many_private(StringFunction::ExtractAll.into(), &[pat], false)
+            .map_many_private(StringFunction::ExtractAll.into(), &[pat], true, false)
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/string.rs:126]: `false`

File: crates/polars-plan/src/dsl/string.rs [hunk 9]
```diff
@@ -322,6 +335,7 @@ impl StringNameSpace {
         self.0.map_many_private(
             FunctionExpr::StringExpr(StringFunction::StripSuffix),
             &[suffix],
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/string.rs:338]: `false`

File: crates/polars-plan/src/dsl/string.rs [hunk 10]
```diff
@@ -313,6 +325,7 @@ impl StringNameSpace {
         self.0.map_many_private(
             FunctionExpr::StringExpr(StringFunction::StripPrefix),
             &[prefix],
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/string.rs:328]: `false`

File: crates/polars-plan/src/dsl/string.rs [hunk 11]
```diff
@@ -281,6 +292,7 @@ impl StringNameSpace {
             FunctionExpr::StringExpr(StringFunction::Replace { n: -1, literal }),
             &[pat, value],
             true,
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/string.rs:295]: `false`

File: crates/polars-plan/src/dsl/string.rs [hunk 12]
```diff
@@ -271,6 +281,7 @@ impl StringNameSpace {
             FunctionExpr::StringExpr(StringFunction::Replace { n, literal }),
             &[pat, value],
             true,
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/string.rs:284]: `false`

File: crates/polars-plan/src/dsl/string.rs [hunk 13]
```diff
@@ -261,6 +270,7 @@ impl StringNameSpace {
             FunctionExpr::StringExpr(StringFunction::Replace { n: 1, literal }),
             &[pat, value],
             true,
+            true,
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/string.rs:273]: `false`

File: crates/polars-plan/src/dsl/string.rs [hunk 14]
```diff
@@ -205,13 +214,13 @@ impl StringNameSpace {
     /// Split the string by a substring. The resulting dtype is `List<Utf8>`.
     pub fn split(self, by: Expr) -> Expr {
         self.0
-            .map_many_private(StringFunction::Split(false).into(), &[by], false)
+            .map_many_private(StringFunction::Split(false).into(), &[by], true, false)
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/string.rs:217]: `false`

File: crates/polars-plan/src/dsl/string.rs [hunk 15]
```diff
@@ -205,13 +214,13 @@ impl StringNameSpace {
...
     }
 
     /// Split the string by a substring and keep the substring. The resulting dtype is `List<Utf8>`.
     pub fn split_inclusive(self, by: Expr) -> Expr {
         self.0
-            .map_many_private(StringFunction::Split(true).into(), &[by], false)
+            .map_many_private(StringFunction::Split(true).into(), &[by], true, false)
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/string.rs:223]: `false`

**Review by ritchie46 — COMMENTED** [review]
I am not sure about all of them. But I believe many of them should be `false`. I trickled them somewhere. If the tests agree, I think we should toggle them. :)

**Comment by ritchie46** [comment]
> The main problem seems to be in the `list namespace`. If there is only one element in each group(e.g. from `groupby` + `implode`), these list operations will produce a `list of list`, but our test case expects it to be un-nested.

Ok, then those need to be `auto-exploded`.

**Comment by ritchie46** [comment]
On it!

**Comment by stinodego** [comment]
This should have been rebased - it broke the main branch 😕

**Pull request merged**